consensus/istanbul/types_v2_test.go

```diff
@@ -183,3 +188,188 @@ func TestRoundChangeV2RLPEncoding(t *testing.T) {
+		Root:        common.BigToHash(common.Big1),
+		TxHash:      common.BigToHash(common.Big256),
+		ReceiptHash: common.BigToHash(common.Big3),
+		Bloom:       [types.BloomByteLength]byte{},
+		Number:      big.NewInt(37),
+		GasUsed:     123213,
+		Time:        100,
+	}
+	fillF(header.Bloom[:], types.BloomByteLength)
+	writeExtra(header, validators)
+	return header
+}
+
+func bigTxs(gasLimit int, gasPerByte int) []*types.Transaction {
+	curr := common.BytesToAddress(getF(20))
+	gateFeeRec := common.BytesToAddress(getF(20))
+	tx := types.NewTransaction(999, common.BytesToAddress(getF(20)),
+		big.NewInt(329274), 2942729, big.NewInt(294279),
+		&curr,
+		&gateFeeRec, big.NewInt(32027), getF(gasLimit/gasPerByte))
+	return []*types.Transaction{tx}
+}
+
+func bigBlock(gasLimit int, validators int, gasPerByte int) *types.Block {
+	randomness := &types.Randomness{
+		Revealed:  common.BigToHash(common.Big3),
+		Committed: common.BigToHash(common.Big2),
+	}
+	// Full header
+	header := bigHeader(validators)
+	txs := bigTxs(gasLimit, gasPerByte)
+	receipts := []*types.Receipt{{
+		Type:              9,
+		PostState:         getF(32),
+		Status:            9,
+		CumulativeGasUsed: 1_729_919,
+		Bloom:             [types.BloomByteLength]byte{},
+		Logs: []*types.Log{{
+			Address: common.BytesToAddress(getF(20)),
+			Topics:  []common.Hash{common.BytesToHash(getF(32)), common.BytesToHash(getF(32))},
+			Data:    getF(53),
+		}},
+	}}
+	fillF(receipts[0].Bloom[:], types.BloomByteLength)
+	block := types.NewBlock(header, txs, receipts, randomness, newHasher())
+	return block
+}
+
+func bigPC(quorum int) *PreparedCertificateV2 {
+	// while quorum amount of commits is imposible by consensus design,
+	// and the max should be (quorum - 1), for simplification
+	// we'll just use quorum commit messages
+	pc := &PreparedCertificateV2{}
+	pc.ProposalHash = common.BytesToHash(getF(32))
+	pc.PrepareOrCommitMessages = make([]Message, quorum)
+	for i := 0; i < quorum; i++ {
+		msg := NewCommitMessage(&CommittedSubject{
+			Subject: &Subject{
+				View:   &View{Round: big.NewInt(21), Sequence: big.NewInt(3_192_191_242)},
+				Digest: common.BytesToHash(getF(32)),
+			},
+			EpochValidatorSetSeal: getF(48),
+			CommittedSeal:         getF(48),
+		}, common.BytesToAddress(getF(20)))
+		msg.Signature = getF(64)
+		pc.PrepareOrCommitMessages[i] = *msg
+	}
+	return pc
+}
+
+func bigRequest(quorum int) *RoundChangeRequest {
+	r := &RoundChangeRequest{
+		Address: common.BytesToAddress(getF(20)),
+		View: View{
+			Round:    big.NewInt(23),
+			Sequence: big.NewInt(2_912_119_182),
+		},
+		PreparedCertificateV2: *bigPC(quorum),
+		Signature:             getF(64),
+	}
+	return r
+}
+
+func bigRCC(validators int) *RoundChangeCertificateV2 {
+	rcc := &RoundChangeCertificateV2{}
+	quorum := int(math.Ceil(float64(2*validators) / 3))
+	rcc.Requests = make([]RoundChangeRequest, quorum)
+	for i := 0; i < quorum; i++ {
+		rcc.Requests[i] = *bigRequest(quorum)
+	}
+	return rcc
+}
+
```

Comment:
```suggestion
// Constructs the largest possible consensus message in order to ensure that it
// does not exceed the built in size limits. Note that byte arrays are filled
// with the byte 0xFF, because that takes more space in RLP encoding than byte
// values < 128.
```

Reply:
Byte arrays are serialized with the 'string' logic, not the 'list' logic of RLP. So the value actually doesn't matter I think in most cases, but I used FF since that made it much more clear to the reader.

If you test serialization comparison of an array fill of zero and a different byte array filled with 0xFF, both will take the exact same space except for arrays of length one.

Reply:
(I'm still adding most of the comment)